### PR TITLE
Implement findOne operation

### LINKED ISSUE #89: Feature Request: Implement findOne function!
Fireorm's `.find` always return an array of elements (even if only one document matches the query), but there are situations when we want to do a query and just take the first result. We could do something in the line of `(await query.limit(1).find())[0]` but that's cumbersome and prone to error.

Implement a `findOne` function in fireorm repositories to support this use case.
 

Acceptance Criteria: 
- Must return null if no document was retrieved
- Add `findOne` in `IQueryBuilder`
- Modify `execute` in `IQueryExecutor` to receive a `single: bool` prop
- Edit `AbstractFirestoreRepository` and `BatchFirestoreRepository` to implement the new function
- Add unit tests
- Add integration test in the `simple_repository` file

### PR DESCRIPTION
**Pull request by skneko**
Closes #89.

~~Unit and integration tests are not done yet, but I'm going to give it a shot. Reviews welcome, please.~~

### CODE REVIEW

**Comment by skneko** [comment]
Ready for review. Build is failing because Travis is unable to run the integration tests, due to the lack of a Firestore private key.

File: src/BaseFirestoreRepository.spec.ts [hunk 1]
```diff
@@ -362,6 +362,28 @@ describe('BaseFirestoreRepository', () => {
     });
   });
 
+  describe('findOne', () => {
+    it('must return T', async () => {
+      await bandRepository.runTransaction(async tran => {
```
Comment (wovalle) [inline src/BaseFirestoreRepository.spec.ts]: Can you remove the `runTransaction` from these tests and add another test just for the transaction?

File: src/QueryBuilder.ts [hunk 2]
```diff
@@ -133,4 +133,20 @@ export default class QueryBuilder<T extends IEntity>
...
+          resolve(singleOrNone[0]);
+        } else {
+          resolve(null);
+        }
+      }).catch(e => {
+        reject(e);
+      });
+    });
+  }
 }
```
Comment (wovalle) [inline src/QueryBuilder.ts]: Great thinking by calling the `execute` method! I think this method can be written in a better way by using the power of `async/await`. Something in the lines of 

```typescript
  async findOne(): Promise<T | null> {
    const queryResult = await this.executor.execute(this.queries, this.limitVal, this.orderByObj, true);

    return queryResult.length ? queryResult[0] : null;
}
```

File: test/functional/1-simple_repository.spec.ts [hunk 3]
```diff
@@ -56,6 +56,12 @@ describe('Integration test: Simple Repository', () => {
       .find();
     expect(byWebsite[0].id).to.equal('dream-theater');
 
+    // Find one band matching some criteria
+    const byWebsiteOne = await bandRepository
```
Comment (wovalle) [inline test/functional/1-simple_repository.spec.ts]: good!

**Comment by wovalle** [comment]
And regarding the integration tests, no worries, I'll deal with the failing test when this pr is ready to merge.

**Comment by skneko** [comment]
I've included your requested changes. Thanks for being so nice, this is one of my first PRs and my first PR of this size :)

**Comment by wovalle** [comment]
@all-contributors please add @skneko for code and ideas

**Pull request merged**